packages/lib/models/Tag.test.js

```diff
@@ -156,4 +156,38 @@ describe('models/Tag', function() {
 		expect(commonTagIds.includes(tagc.id)).toBe(true);
 	}));
 
+	it('should sort tags for setTags, Sidebar and TagList', (async () => {
+		const folder1 = await Folder.save({ title: 'folder1' });
+		const note1 = await Note.save({ title: 'ma note 0', parent_id: folder1.id });
+
+		// original/sorted order
+		const tag1 = await Tag.save({ title: '#coding' });
+		const tag2 = await Tag.save({ title: '#house' });
+		const tag3 = await Tag.save({ title: '#wait' });
+		const tag4 = await Tag.save({ title: '@⏲15 min' });
+		const tag5 = await Tag.save({ title: '@⏲30 min' });
+		const tag6 = await Tag.save({ title: '@⏲60 min' });
+
+		await Tag.addNote(tag4.id, note1.id);
+		await Tag.addNote(tag2.id, note1.id);
+		await Tag.addNote(tag1.id, note1.id);
+		await Tag.addNote(tag6.id, note1.id);
+		await Tag.addNote(tag3.id, note1.id);
+		await Tag.addNote(tag5.id, note1.id);
+
+		// This test checks the sorting functionality for tags in all three files(setTags, Sidebar and TagList)
+		// commonTags contains unsortedTags
+		const commonTags = await Tag.commonTagsByNoteIds([note1.id]);
```

Comment:
That doesn't do what you say it's doing. With the way it's implemented now, you don't need to test setTags or the SideBar, so please remove this test.